packages/tentales/src/types/index.d.ts

```diff
@@ -0,0 +1,72 @@
+import * as Koa from "koa"
```

Comment:
Imo it would be better to place the types in proper locations. IMO `types/*.d.ts` should be used when type definitions are missing

Reply:
Nice comment! This was actually one of the major things I wanted to discuss, but totally forgot about.

Looking at the `packages/tentales/src/types/index.d.ts` file, there's a handful of type/interface declarations there, which I then import and use in other files. My reasoning behind placing the declarations in this file, and not in a normal module file, is because they're shared and used by multiple modules. 

Question here is. Is this the way to go? If not, where to put them?

@palmfjord @jalet @janneh @BeeeQueue

Reply:
I rather have the interface with the module instad of another file

Reply:
I think they should be in a module rather than in the global scope. Or at least a namespace.

Reply:
These are not global though. There's no ts compiler `types` configuration draging them into the global scope - they're included as `import { ServiceCaller, ServiceConfig, Action } from "../types"`

Reply:
ehm right. sorry 😂 i don't understand why it needs to be a `d.ts` file though? i feel like that's a little confusing

Reply:
It doesn't... just me not really knowing what to do with these shared type declarations ;) But I'll figure it out

Reply:
Coolio 🆒